pkg/kubectl/cmd/create_role.go

```diff
@@ -175,10 +201,17 @@ func (c *CreateRoleOptions) RunCreateRole(f cmdutil.Factory, cmdOut io.Writer, c
 	// 2. Prevents pointing to non-existent resources.
 	// 3. Transfers resource short name to long name. E.g. rs.extensions is transferred to replicasets.extensions
 	for _, r := range c.Resources {
-		resource, err := mapper.ResourceFor(r)
+		resource, err := mapper.ResourceFor(schema.GroupVersionResource{Resource: r.Resource, Group: r.Group})
 		if err != nil {
 			return err
 		}
+		if len(r.SubResource) > 0 {
+			sub, err := mapper.ResourceFor(schema.GroupVersionResource{Resource: r.SubResource})
```

Comment:
`ResourceFor` here  can get the full name of subresource. just like `scale` to `scales`